src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityReportListenerImpl.java

```diff
@@ -63,19 +63,34 @@ public void onReport(DockerTraceabilityReport report) {
     private void processReport(@Nonnull DockerTraceabilityReport report) throws IOException {
         DockerTraceabilityPlugin plugin = DockerTraceabilityPlugin.getInstance();
 
-        // Get fingerprints for the image
-        Fingerprint imageFP = DockerFingerprints.of(report.getImageId());
-        if (imageFP == null && plugin.getConfiguration().isCreateImageFingerprints()) {
-            LOGGER.log(Level.FINE, "Creating a new fingerprint for image {0}", report.getImageId());
-            imageFP = DockerTraceabilityHelper.makeImage(report.getImageId(), 
-                    report.getImageName(), report.getEvent().getTime());
+        String imageId = report.getImageId();
+        if (imageId == null) { // Try to restore imageId by container id
+            String containerId = report.getContainerId();
+            if (containerId != null) {
+                final Fingerprint containerFP = DockerFingerprints.of(containerId);
+                final DockerDeploymentFacet facet = containerFP != null
+                        ? DockerFingerprints.getFacet(containerFP, DockerDeploymentFacet.class)
+                        : null;
+                imageId = facet != null ? facet.getImageId() : null;
+            }
         }
         
-        if (imageFP == null) { // We don't create anything and exit
-            LOGGER.log(Level.FINE, "Cannot get or create a fingerprint for image {0}. "
-                + "Most probably, the image has not been created in Jenkins. Report will be ignored", 
-                    report.getImageId());
-            return;
+        // Get fingerprints for the image
+        Fingerprint imageFP = null;
+        if (imageId != null) {
+            imageFP = DockerFingerprints.of(imageId);
+            if (imageFP == null && plugin.getConfiguration().isCreateImageFingerprints()) {
+                LOGGER.log(Level.FINE, "Creating a new fingerprint for image {0}", report.getImageId());
+                imageFP = DockerTraceabilityHelper.makeImage(imageId, 
+                        report.getImageName(), report.getEvent().getTime());
+            }
+        
+            if (imageFP == null) { // We don't create anything and exit
+                LOGGER.log(Level.FINE, "Cannot get or create a fingerprint for image {0}. "
+                    + "Most probably, the image has not been created in Jenkins. Report will be ignored", 
+                        report.getImageId());
+                return;
+            }
         }
```

Comment:
imageId may be null for a container that is being tracked by jenkins (the image was build by Jenkins - but the container deployment/stop/rm cycle happened when the bit sending the reports wasn't running)
I would have expected a fallback to imageName - although for multiple docker servers this may be non-unique :(